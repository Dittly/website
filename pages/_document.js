import Document, {Head, Main, NextScript} from 'next/document'
import {ServerStyleSheet} from 'styled-components'

// Adds normalize and general styles
import '../styles/base-styles'

export default class MyDocument extends Document {
  render() {
    const sheet = new ServerStyleSheet()
    const main = sheet.collectStyles(<Main />)
    const styleTags = sheet.getStyleElement()
    return (
      <html lang="en">
        <Head>
          <meta
            name="google-site-verification"
            content="wjFi2JtlRC-jy5mCnCc2epXwTHGfsYWfPQB7lkubC_U"
          />
          <title>Dittly</title>

          <meta
            name="viewport"
            content="width=device-width, height=device-height, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
          />
          <link
            href="https://fonts.googleapis.com/css?family=Arvo|Rubik:400,500"
            rel="stylesheet"
          />
          <link
            href="https://fonts.googleapis.com/css?family=Roboto+Slab"
            rel="stylesheet"
          />

          {styleTags}
        </Head>
        <body>
          <div className="root">{main}</div>
          <NextScript />
        </body>
      </html>
    )
  }
}
