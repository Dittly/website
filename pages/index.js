import MainLayout from '../components/layouts/main'
import withData from '../utils/apollo/with-data'

const Index = (props) => (
  <MainLayout>
    Homepage
  </MainLayout>
)

export default withData(Index)
