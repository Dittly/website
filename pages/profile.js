import BasePageComponent from '../components/base-page'
import withAuth from '../components/with-auth'
import Layout from '../components/layouts'
import ProfileContent from '../src/profile/index'

export class Profile extends BasePageComponent {
  render() {
    return (
      <Layout>
        <ProfileContent {...this.props} />
      </Layout>
    )
  }
}

export default withAuth(Profile)
