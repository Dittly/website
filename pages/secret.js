import React from 'react'
import PropTypes from 'prop-types'

import MainLayout from '../components/layouts/main'
import withAuth from '../components/with-auth'

const Secret = ({loggedInUser, signOut}) => (
  <MainLayout>
    Hello {loggedInUser.user.name}!<br />
    <button onClick={signOut}>Sign out</button>
  </MainLayout>
)

Secret.propTypes = {
  signOut: PropTypes.func.isRequired,
  loggedInUser: PropTypes.shape({
    user: PropTypes.shape({
      name: PropTypes.string.isRequired
    })
  })
}

export default withAuth(Secret)
