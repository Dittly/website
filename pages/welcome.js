import BasePageComponent from '../components/base-page'
import withAuth from '../components/with-auth'
import Layout from '../components/layouts'
import WelcomeContent from '../src/welcome'

export class Welcome extends BasePageComponent {
  render() {
    return (
      <Layout>
        <WelcomeContent {...this.props} />
      </Layout>
    )
  }
}

export default withAuth(Welcome)
