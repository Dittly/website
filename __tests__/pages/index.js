/* eslint-env jest */
import {mount} from 'enzyme'

import {Index} from '../../pages'

describe('pages/index', () => {
  it(`verifies ${Index.name} renders correctly`, () => {
    const wrapper = mount(<Index />)
    expect(wrapper.length).toBe(1)
  })
})
