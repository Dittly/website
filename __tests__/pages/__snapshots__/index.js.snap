// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page renders / correctly 1`] = `
<div>
  <header>
    <div
      className="box-s1qf175f-0 iNmKap"
      color="white"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Dittly
      </a>
       | 
      <a
        href="/secret"
        onClick={[Function]}
      >
        Secret
      </a>
       | 
      <a
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </header>
  Homepage
</div>
`;
